netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -288,16 +309,25 @@ private void sendGrpcFrame(ChannelHandlerContext ctx, SendGrpcFrameCommand cmd,
    * Handler for a GOAWAY being either sent or received.
    */
   private void goingAway() {
-    Status goAwayStatus = goAwayStatus();
+    final Status goAwayStatus = goAwayStatus();
     if (connection().goAwayReceived()) {
       // Received a GOAWAY from the remote endpoint. Fail any streams that were created after the
       // last known stream.
-      int lastKnownStream = connection().local().lastKnownStream();
-      for (Http2Stream stream : http2Streams()) {
-        if (lastKnownStream < stream.id()) {
-          clientStream(stream).transportReportStatus(goAwayStatus, false, new Metadata.Trailers());
-          stream.close();
-        }
+      final int lastKnownStream = connection().local().lastKnownStream();
+      try {
+        connection().forEachActiveStream(new Http2StreamVisitor() {
+          @Override
+          public boolean visit(Http2Stream stream) throws Http2Exception {
+            if (lastKnownStream < stream.id()) {
+              clientStream(stream).transportReportStatus(goAwayStatus, false,
+                  new Metadata.Trailers());
+              stream.close();
```

Comment:
@buchgr I suppose it doesn't matter too much, but I think I'd prefer to keep this logic together for clarity. I have simplified this method a bit since it doesn't matter if GOAWAY is sent or received.